### DISCUSSION TITLE
Swapping windows question

### DISCUSSION THREAD

**Discussion by knauth** [i3/i3#5948] [Q&A] [answered]
Is there any way to establish a sort of top-level container? I want a big window on the far left and everything to tile vertically in a column to the right of it, and I’d like to be able to swap windows in and out of this left area with one keybind. Is there any way to do this that doesn’t involve remarking the window every time?

**Comment by okraits** [comment]
If you have a split container on the left (it doesn't matter wether it has a tiled, tabbed or stacked layout), you can add and remove windows by moving them to the left or right. If you want to be able to toggle that (one command for both adding and removing windows), you could write a small script using the IPC interface.

**Comment by orestisfl** [comment, marked as answer]
The simplest i3 command sequence to do that without IPC (you will have to create the setup manually) is: `swap with mark _left, mark _left`.

You could otherwise write a simple script that arranges all windows of the current workspace in the layout you want as a one-time setup.

**Reply by knauth** [reply]
This is perfect, thanks.
